python/src/scipp/plot/profiler.py

```diff
@@ -0,0 +1,340 @@
+                    self.slice_pos_rectangle.set_x(new_pos)
+                    if self.histograms[self.name][dim][dim]:
+                        self.slice_pos_rectangle.set_width(self.slider_coord[
+                            self.name][dim][dim, val.value + 1].value -
+                                                           new_pos)
+                    else:
+                        new_pos -= 0.5 * self.slice_pos_rectangle.get_width()
+                    self.slice_pos_rectangle.set_x(new_pos)
+
+    def toggle_mask(self, change):
+        super().toggle_mask(change)
+        if self.profile_viewer is not None:
+            self.profile_viewer[self.profile_key].masks[self.name][
+                change["owner"].masks_name].value = change["new"]
+        return
+
+    def compute_profile(self, event):
+        # Find indices of pixel where cursor lies
+        dimx = self.xyrebin["x"].dims[0]
+        dimy = self.xyrebin["y"].dims[0]
+        ix = int((event.xdata - self.current_lims["x"][0]) /
+                 self.image_pixel_size[dimx])
+        iy = int((event.ydata - self.current_lims["y"][0]) /
+                 self.image_pixel_size[dimy])
+        # Resample the 3d cube down to a 1d profile
+        return self.resample_image(self.da_with_edges,
+                                   coord_edges={
+                                       dimy: self.da_with_edges.coords[dimy],
+                                       dimx: self.da_with_edges.coords[dimx]
+                                   },
+                                   rebin_edges={
+                                       dimy: self.xyrebin["y"][dimy,
+                                                               iy:iy + 2],
+                                       dimx: self.xyrebin["x"][dimx, ix:ix + 2]
+                                   })[dimy, 0][dimx, 0]
+
+    def create_profile_viewer(self, prof):
+        # We need to extract the data again and replace with the original
+        # coordinates, because coordinates have been forced to be bin-edges
+        # so that rebin could be used. Also reset original unit.
+        to_plot = sc.DataArray(data=sc.Variable(dims=prof.dims,
+                                                unit=self.data_array.unit,
+                                                values=prof.values,
+                                                variances=prof.variances))
+        for dim in prof.dims:
+            to_plot.coords[dim] = self.slider_coord[self.name][dim]
+        if len(prof.masks) > 0:
+            for m in prof.masks:
+                to_plot.masks[m] = prof.masks[m]
+        self.profile_viewer = plot({self.name: to_plot},
+                                   ax=self.ax_pick,
+                                   logy=self.log)
+        self.profile_key = list(self.profile_viewer.keys())[0]
+        return to_plot
+
+    def update_profile(self, event):
+        if event.inaxes == self.ax:
```

Comment:
Probably a matter of taste, but I find long `if` blocks hard to read, especially in Python that does not have `}` to mark the end. Could be avoided using something like:
```python
if event.inaxes != self.ax:
  return
# main part of the function with 1 indent level fewer
```